pkg/pod/pod.go

```diff
@@ -599,6 +599,49 @@ func (builder *Builder) WithPrivilegedFlag() *Builder {
 	return builder
 }
 
+// WithVolume attaches given volume to all pod's containers.
+func (builder *Builder) WithConfigMapVolume(volumeName, cmName string, defMode *int32) *Builder {
```

Comment:
Since Volumes support a large amount of sources, can you just make this generic? Like `WithVolume` that accepts a v1.Volume?

Reply:
I think it should be pretty similar to the WithVolumeMount method

Reply:
Indeed, from what I know `WithLocalVolume` does use configMap.

What is does not do, is setting the `defaultMode`. 
If that is what you want, changing the signature of this function could break existing usage of that function.

Reply:
No `WithConfigMapVolume` should work only with _ConfigMap_ volume. `WithLocalVolume` does extra things e.g. adding _VolumeMount_ to all the containers or doesn't allow configuration of the _VolumeMount_'s name/path.

Based on @trewest comment I'll check creating of a generic method that will add _Volume_ (despite type) to the _Pod_

Reply:
@trewest updated